client/components/UserDecalPanel.js

```diff
@@ -87,6 +90,37 @@ class UserDetailsPanel extends Component {
     return null
   }
 
+  _renderDecalCode = (decal, index) => (
```

Comment:
render methods should be "public" (without the `_`), and placed directly above the main render method.